src/Sulu/Bundle/AudienceTargetingBundle/Controller/TargetGroupController.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AudienceTargetingBundle\Controller;
+
+use Doctrine\ORM\EntityManagerInterface;
+use FOS\RestBundle\Controller\Annotations\Delete;
+use FOS\RestBundle\Controller\Annotations\Get;
+use FOS\RestBundle\Controller\Annotations\RouteResource;
+use FOS\RestBundle\Routing\ClassResourceInterface;
+use Sulu\Bundle\AudienceTargetingBundle\Entity\TargetGroupInterface;
+use Sulu\Bundle\AudienceTargetingBundle\Entity\TargetGroupRepositoryInterface;
+use Sulu\Bundle\AudienceTargetingBundle\Mapper\TargetGroupMapperInterface;
+use Sulu\Component\Rest\Exception\EntityNotFoundException;
+use Sulu\Component\Rest\Exception\MissingParameterException;
+use Sulu\Component\Rest\ListBuilder\FieldDescriptorInterface;
+use Sulu\Component\Rest\ListBuilder\ListRepresentation;
+use Sulu\Component\Rest\RestController;
+use Sulu\Component\Security\SecuredControllerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Makes target groups available through a REST API.
+ *
+ * @RouteResource("target-group")
+ */
+class TargetGroupController extends RestController implements ClassResourceInterface, SecuredControllerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected static $entityKey = 'target-groups';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSecurityContext()
+    {
+        return 'sulu.settings.target-groups';
+    }
+
+    /**
+     * Returns all fields that can be used by list.
+     *
+     * @Get("target-groups/fields")
+     *
+     * @return Response
+     */
+    public function getFieldsAction()
+    {
+        return $this->handleView($this->view($this->getFieldDescriptors()));
+    }
+
+    /**
+     * Returns list of target-groups.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        $restHelper = $this->get('sulu_core.doctrine_rest_helper');
+        $factory = $this->get('sulu_core.doctrine_list_builder_factory');
+        $listBuilder = $factory->create($this->getTargetGroupRepository()->getClassName());
+
+        $fieldDescriptors = $this->getFieldDescriptors();
+        $restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
+
+        $fieldsParam = $request->get('fields');
+        $fields = explode(',', $fieldsParam);
+        if (!$fieldsParam || false !== array_search('webspaceKeys', $fields)) {
```

Comment:
the `OR` is not avoidable, since the one OR the other statement needs to be fulfilled.